packages/discovery-provider/plugins/pedalboard/apps/trending-challenge-rewards/src/rewards.ts

```diff
@@ -0,0 +1,139 @@
+import { App } from '@pedalboard/basekit'
+import { Knex } from 'knex'
+import { Ok, Err, Result } from 'ts-results'
+import { SharedData } from './config'
+import {
+  getChallengesDisbursementsUserbanksFriendlyEnsureSlots,
+  getTrendingChallenges,
+  getTrendingChallengesByDate
+} from './queries'
+import { WebClient } from '@slack/web-api'
+import { formatDisbursementTable } from './slack'
+import { discoveryDb } from './utils'
+import { ChallengeId } from '@audius/sdk'
+import axios from 'axios'
+
+type Challenge = {
+  challenge_id: string
+  user_id: string
+  specifier: string
+  amount: string
+  completed_blocknumber: number
+  handle: string
+  wallet: string
+}
+
+
+// TODO: move something like this into App so results are commonplace for handlers
+export const disburseTrendingRewards = async (
+  app: App<SharedData>
+): Promise<void> => {
+  const { dryRun } = app.viewAppData()
+  const disburse = await onDisburse(app, dryRun)
+  disburse.mapErr(console.error)
+}
+
+// dryRun separated out so it can be run manually via another trigger
+export const onDisburse = async (
+  app: App<SharedData>,
+  dryRun: boolean,
+  targetSpecifier?: string
+): Promise<Result<undefined, string>> => {
+  const db = discoveryDb
+  const { slackBotToken, slackChannel } = app.viewAppData()
+  if (slackBotToken === undefined) return new Err('SLACK_BOT_TOKEN undefined')
+  const client = new WebClient(slackBotToken)
+
+  console.log(`doing ${dryRun ? 'a dry run' : 'the real deal'}`)
```

Comment:
"THE REAL DEAL".
But also was this for testing or do we wan't to keep it in the output for debugging purposes?

Reply:
not my code lol, but yeah i can change this